pyi.py

```diff
@@ -249,21 +243,38 @@ def handleNodeDelete(self, node: ast.AST) -> None:
 
 
 class PyiAwareFileChecker(checker.FileChecker):
-    def run_check(self, plugin: Plugin, **kwargs: Any) -> Any:
-        if self.filename == "-":
-            filename = self.options.stdin_display_name
-        else:
-            filename = self.filename
+    if FLAKE8_MAJOR_VERSION < 5:
 
-        if filename.endswith(".pyi") and plugin["plugin"] == FlakesChecker:
-            LOG.info(
-                "Replacing FlakesChecker with PyiAwareFlakesChecker while "
-                "checking %r",
-                filename,
-            )
-            plugin = dict(plugin)
-            plugin["plugin"] = PyiAwareFlakesChecker
-        return super().run_check(plugin, **kwargs)
+        def run_check(self, plugin, **kwargs: Any) -> Any:
+            if self.filename == "-":
+                filename = self.options.stdin_display_name
+            else:
+                filename = self.filename
+
+            if filename.endswith(".pyi") and plugin["plugin"] is FlakesChecker:
+                LOG.info(
+                    f"Replacing FlakesChecker with PyiAwareFlakesChecker while "
+                    f"checking {filename!r}",
+                )
+                plugin = dict(plugin)
+                plugin["plugin"] = PyiAwareFlakesChecker
+            return super().run_check(plugin, **kwargs)
+
+    else:
+
+        def run_check(self, plugin, **kwargs: Any) -> Any:
+            if self.filename == "-":
+                filename = self.options.stdin_display_name
+            else:
+                filename = self.filename
+
+            if filename.endswith(".pyi") and plugin.obj is FlakesChecker:
+                LOG.info(
+                    f"Replacing FlakesChecker with PyiAwareFlakesChecker while "
+                    f"checking {filename!r}",
+                )
+                plugin = plugin._replace(obj=PyiAwareFlakesChecker)
```

Comment:
Where does `_replace` come from? I can't find it in the flake8 source.

Reply:
It's a public attribute of `namedtuple`s: https://docs.python.org/3/library/collections.html#collections.namedtuple